lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -744,8 +744,18 @@ public function computeChangeSet(ClassMetadata $class, $entity)
 
                 $orgValue = $originalData[$propName];
 
-                if ($orgValue instanceof BackedEnum) {
-                    $orgValue = $orgValue->value;
+                if (! empty($class->fieldMappings[$propName]['enumType'])) {
```

Comment:
if somebody is interested why I added this check, its because I want to avoid the case where we have an array and go over the whole array checking if its `BackedEnum` for no reason. This way, unless there are enums in the play for the given field, no checks whatsoever are performed. Likely 0 performance gains with simple enums, but large performance gain with arrays of enums.